alf/networks/encoding_networks.py

```diff
@@ -230,6 +230,149 @@ def __init__(self,
         super().__init__(nets, input_tensor_spec=input_tensor_spec, name=name)
 
 
+@alf.configurable
+class ImageDeconvNetwork(_Sequential):
+    """
+    A general template class for creating image deconv (transposed convolutional)
+        networks with auto-shape inference.
+    """
+
+    def __init__(self,
+                 input_size,
+                 transconv_layer_params,
+                 output_shape,
+                 start_decoding_channels,
+                 preprocess_fc_layer_params=None,
+                 activation=torch.relu_,
+                 kernel_initializer=None,
+                 output_activation=torch.tanh,
+                 name="ImageDeconvNetwork"):
+        """
+        Instead of specifying an initial start shape for image deconv, this
+        class only needs to specify the desired output shape for the image
+        and will automatically calculate the desired shape to start
+        decoding based on the specified ``transconv_layer_params``
+        and uses a FC layer to map the to the desired start shape.
+
+        Args:
+            input_size (int): the size of the input latent vector
+            transconv_layer_params (tuple[tuple]): a non-empty
+                tuple of tuple (num_filters, kernel_size, strides, padding),
+                where ``padding`` is optional.
+            output_shape (tuple): the complete output size would be
+                output_shape = (c, h, w).
+            start_decoding_channels (int): the initial number of channels we'd
+                like to have for the feature map. Note that we always first
+                project an input latent vector into a vector of an appropriate
+                length so that it can be reshaped into (``start_decoding_channels``,
+                ``start_decoding_height``, ``start_decoding_width``).
```

Comment:
There are no longer input ``start_decoding_height`` and ``start_decoding_width``.

Reply:
Added comments on them.